controllers/buildpipeline/buildpipeline_adapter.go

```diff
@@ -108,6 +108,36 @@ func (a *Adapter) EnsureSnapshotExists() (result controller.OperationResult, err
 		return controller.ContinueProcessing()
 	}
 
+	if _, found := a.pipelineRun.ObjectMeta.Annotations[tekton.SnapshotNameLabel]; found {
```

Comment:
what if this annotation points to a non-existent snapshot?

Reply:
I thought about that too, but the Snapshot could have been pruned or deleted by the user. The annotation is there to tell us that a single Snapshot was already created for that PLR, not that the Snapshot should still exist.

I think that we shouldn't create a new Snapshot after we created it once, otherwise we could start recreating Snapshots for old(er) pipelineRuns which I think is unexpected behavior.